pkg/kubectl/cmd/expose.go

```diff
@@ -55,6 +58,14 @@ var (
 		--port and the exposed resource has multiple ports, all will be re-used by the new service. Also if no 
 		labels are specified, the new service will re-use the labels from the resource it exposes.
 
+		Note that if you want your exposed resouce controller to be rolling-upgradeable,
+		the services need to contain at least two labels, one of which will not be targeted
+		by the selector of your service. This is because at least one of the labels needs to change
```

Comment:
Did not find it in the docs. This is how it behaves. The problem is described in #17902